### PR TITLE
Add support for `mini.pick` as a picker

### PR DESCRIPTION
**Pull request by pkazmier**
With [mini](https://github.com/echasnovski/mini.nvim/tree/main) gaining in popularity, this commit adds support for [mini.pick](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-pick.md).

Here is a short screencast showing this new picker in action with Zk notes:

https://github.com/zk-org/zk-nvim/assets/747855/cf645482-c3eb-4713-8925-bb1501f54cb8

### CODE REVIEW

File: lua/zk/pickers/minipick.lua
Comment (jurica) [inline lua/zk/pickers/minipick.lua:5]: Tbh I don't really like that comment. The other picker implementations don't have "interface implementation" commentend either and I don't really see any added value in them.

File: lua/zk/pickers/minipick.lua
Comment (jurica) [inline lua/zk/pickers/minipick.lua:8]: Same as above.

File: lua/zk/pickers/minipick.lua
Comment (jurica) [inline lua/zk/pickers/minipick.lua:34]: Same as above.

**Review by jurica — COMMENTED** [review]
Hi @pkazmier , that's very cool thank you for the contribution!
Apart from some minor nitpicks the code looks good to me. But I would be glad if @kabouzeid could also take a look.

@zk-org/maintainers how do we want to handle this contribution? As we're still in maintenance mode strictly taken we shouldn't merge it yet. On the other hand there isn't really a release process for zk-nvim currently.

**Comment by tjex** [comment]
Although we've gotten through all the bugs for `zk-nvim`, I would consider us still in maintenance mode ([as per the readme](https://github.com/zk-org/zk-nvim?tab=readme-ov-file#news-we-are-in-maintenance-mode)) until we cut a new release of `zk`. This should realistically be quite soon, at which point we may want to look at starting with release versioning with `zk-nvim` as well. 

I would say we should continue holding off on all feature request PRs until the above has happened?

File: lua/zk/pickers/minipick.lua
Comment (pkazmier) [reply lua/zk/pickers/minipick.lua:5]: Agreed. That was a comment I added to show the maintainer of mini how I integrated it here. I’ll remove it.

File: README.md
Comment (kabouzeid) [inline README.md:77]: what about naming it `mini.pick` or `mini` instead?

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [inline lua/zk/pickers/minipick.lua:20]: what does this do?

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [inline lua/zk/pickers/minipick.lua:59]: does mini not support multi selection?

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [inline lua/zk/pickers/minipick.lua:69]: what is that for?

**Review by kabouzeid — COMMENTED** [review]
Overall looks good to me and tried it out locally. Please clarify the code though.

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [reply lua/zk/pickers/minipick.lua:59]: I think it does. It's called "mark" in mini.pick.

File: README.md
Comment (pkazmier) [reply README.md:77]: Sure, I'll rename it to `mini.pick`.

File: lua/zk/pickers/minipick.lua
Comment (pkazmier) [reply lua/zk/pickers/minipick.lua:20]: The `choose` function in `mini.pick` is used to take action on an item—such as opening a buffer to edit a file. Because `zk-nvim` already handles what to do with selected options, I set this to `nil`. The value selected is returned as a result of `MiniPick.start()`. I then pass that value to the provided `zk-nvim` callback to handle the "action" of what to do.

File: lua/zk/pickers/minipick.lua
Comment (pkazmier) [reply lua/zk/pickers/minipick.lua:59]: It does support multi-selection, but the `MiniPick.start()` method only returns a single note even if multiple notes were selected. Similar to the above comment about `choose`,`mini.pick` does have a `choose_marked` to take action on multiple selected items. For the same reasons above, I want to delegate to the user supplied callback function, but I don't know how to get access to the ... [scratch that last sentence, as I was typing this comment, I realized I should just be invoking the user-supplied callback in the `choose` functions. Let me play around with this and will update this weekend. Thanks for asking the question].

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [reply lua/zk/pickers/minipick.lua:20]: Why do you need to specify it then? The default is already `choose = nil`. Does that not work for some reason?

File: lua/zk/pickers/minipick.lua
Comment (kabouzeid) [reply lua/zk/pickers/minipick.lua:69]: why only 4 digits? if you really want to limit this number for some reason, I would rather use ">9999" instead of "…999"

File: lua/zk/pickers/minipick.lua
Comment (pkazmier) [reply lua/zk/pickers/minipick.lua:20]: There is a difference between a `nil` value for `choose` and a function that returns `nil`. The former invokes the default action from `mini.pick` while the latter tells `mini.pick` to close the picker without executing an action. But as I mentioned in further comment, I'll be refactoring this.

File: lua/zk/pickers/minipick.lua
Comment (pkazmier) [reply lua/zk/pickers/minipick.lua:69]: When I refactor this later this weekend, there will be no truncation as I'll simply compute the max width required, so the count field will never truncated, so that will eliminate the need to make any decision here.

File: README.md
Comment (pkazmier) [reply README.md:77]: I recall why I named it `minipick`. Because that option is just concatenated to `"zk.pickers"` and then required. So, if we rename that, then we need to rename the file as well, but I don't think you can have dots in the filename, right? And I prefer to name it "minipick" over "mini" as many users only use a subset of the mini umbrella.

**Comment by pkazmier** [comment]
Refactored the code to address all of your comments: multi-select, padding, etc ...

**Review by kabouzeid — APPROVED** [review]

**Comment by kabouzeid** [comment]
thanks for the contribution!

**Comment by pkazmier** [comment]
Thank you!

**Pull request merged**
